internal/models/pubkey_model.go

```diff
@@ -17,3 +28,39 @@ type Pubkey struct {
 	// Public key SHA256 fingerprint. Generated (read-only).
 	Fingerprint string `db:"fingerprint" json:"fingerprint"`
 }
+
+// FingerprintAWS calculates fingerprint used by AWS.
+// AWS calculates MD5 sums differently then all the other tools and use format DER as the base for the sub.
+// see https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/verify-keys.html for more details.
+func (pk *Pubkey) FingerprintAWS() (string, error) {
+	pkey, _, _, _, err := ssh.ParseAuthorizedKey([]byte(pk.Body))
+	if err != nil {
+		return "", fmt.Errorf("failed to calculate fingerprint for %s: %w", pk.Name, err)
+	}
+	// when rsa key
+	if pkey.Type() == "ssh-rsa" {
+		return fingerprintRsaAWS(pkey)
+	}
+	// when ed25519 key
+	// this is the same as what we store, but better be sure here
+	return strings.TrimLeft(ssh.FingerprintSHA256(pkey), "SHA256:") + "=", nil
+}
+
+// fingerprintRsaAWS calculates fingerprint for rsa keys
+func fingerprintRsaAWS(key ssh.PublicKey) (string, error) {
+	parsedCryptoKey := key.(ssh.CryptoPublicKey)
+
+	// Finally, we can convert back to an *rsa.PublicKey
+	pub := parsedCryptoKey.CryptoPublicKey().(*rsa.PublicKey)
+
+	der, err := x509.MarshalPKIXPublicKey(pub)
+	if err != nil {
+		return "", fmt.Errorf("failed to calculate rsa key fingerprint: %w", err)
+	}
+	md5sum := md5.Sum(der) //#nosec
+	hexarray := make([]string, len(md5sum))
+	for i, c := range md5sum {
+		hexarray[i] = hex.EncodeToString([]byte{c})
+	}
+	return strings.Join(hexarray, ":"), nil
+}
```

Comment:
Adding methods into models is something that I would really encourage us to avoid completely. Maybe for very tiny functions where it is obvious these are some kind of field-level small things. This is already very alarming because we now know there are at lest three possible SSH fingerprints (legacy, legacy-DER, SHA) and this can get our of control very quickly.

If we want to go through the "on the fly" generation, then I would rather see some sort of interface like:

```go
type Fingerprinter interface {
 Calculate(pubkey model.Pubkey) (string, error)
}
```

Or something similar. I feel like precalculating those and storing them in the DB is more scalable solution, tho, I realize there is some extra effort in this (we need a migration with existing keys recalculation which is a precedent on this project).